crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -24,6 +25,11 @@ pub mod graph {
 }
 pub const CORE_3D: &str = graph::NAME;
 
+#[cfg(any(not(feature = "webgl"), not(target_arch = "wasm32")))]
+pub const CORE_3D_DEPTH_FORMAT: TextureFormat = TextureFormat::Depth32FloatStencil8;
```

Comment:
This format is poorly supported on android Vulkan (D32_SFLOAT_S8_UINT, always good to check format/feature support on https://vulkan.gpuinfo.org/), so we need to check if it's supported. (and can't make it a const)

https://docs.rs/wgpu/0.17.0/wgpu/struct.Features.html#associatedconstant.DEPTH32FLOAT_STENCIL8

Reply:
Good looking out, I guess I'll need to add this to the various PipelineKey's to switch to other formats. Maybe it should have a few so if someone isn't using deferred they can still use a 32bit format.
Could fit all 4 of the main ones in two bits:
Depth24Plus
Depth24PlusStencil8
Depth32Float
Depth32FloatStencil8

Reply:
I've done the initial work to have this defined at runtime. Currently it defaults to `Depth32Float`, and if there is a camera with deferred it will switch to either `Depth24PlusStencil8` or `Depth32FloatStencil8` depending on what is supported. The user can also pick something by initialing the `Core3DDepthFormat` resource before default plugins. I don't think switching deferred formats after the textures have all been setup works. But I'm not sure switching the depth format after everything is initially setup is a worthwhile feature. Probably need to prove another easy method that auto picks the right format besides having a camera with the `DeferredPrepass` for setting the texture since the camera may not exist initially. I liked being able to avoid needing the user to do anything in `App::new()` to use deferred, tbd if this will end up requiring that. 

I guess another question is if we should support multiple cameras using different depth formats? Seems unlikely that someone would actually need that, but idk.